### PR TITLE
Fix #13908 - x86 aoj for instruction with hidden operand ##asm

### LINKED ISSUE #13908: aoj missing x86/x64 semantic information of instructions
It affects any instruction that has side effects such `push` because opex only shows the visible operands and anything hidden is ignored.
I'm not entirely sure if this is a bug or not so please forgive me if this the intended behaviour.

### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Win7 x64
| File format of the file you reverse (mandatory)      | PE
| Architecture/bits of the file (mandatory)            | x86/32, x86/64
| r2 -v full output, **not truncated** (mandatory)     | radare2 3.5.0-git 21238 @ windows-x86-64 git.3.5.0-git commit: 1c279c6081bc1877fabd533a8314eb527e32c66c build: Wed 04/10/2019__10:52:39.32

### Expected behavior
aoj (opex) should provide information about modified/accessed registers and cpu flags that are hidden.

### Actual behavior
img

### Additional Logs, screenshots, source-code,  configuration dump, ...
The responsible function seem to be opex from `libr/anal/p/anal_x86_cs.c`

### PR DESCRIPTION
**Pull request by x0urc3**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](../DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](../DEVELOPERS.md#code-style)
- [X] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](../../radare2book) with the relevant information (if needed)

**Detailed description**

Fix for aoj on x86 to handle instructions with hidden operand
- Add hidden_op()
- Added operands info for pushf, popf, pushfd, popfd, pushfq, popfq
- Add test for aoj pushf

**Test plan**

Added test cases

**Closing issues**

closes #13908
...

### CODE REVIEW

File: libr/anal/p/anal_x86_cs.c [hunk 1]
```diff
@@ -63,11 +63,58 @@ struct Getarg {
 	int bits;
 };
 
-static void opex(RStrBuf *buf, csh handle, cs_insn *insn) {
+static void hidden_op(csh handle, cs_insn *insn, cs_x86 *x, int mode) {
+	unsigned int id = insn->id;
+	int regsz = 4;
+	switch (mode) {
+		case CS_MODE_64: regsz = 8; break;
```
Comment (radare) [inline libr/anal/p/anal_x86_cs.c]: Wrong indent. Cases go at the same level as switch

File: libr/anal/p/anal_x86_cs.c [hunk 2]
```diff
@@ -63,11 +63,58 @@ struct Getarg {
...
+		case X86_INS_PUSHFD:
+		case X86_INS_POPFD:
+		case X86_INS_PUSHFQ:
+		case X86_INS_POPFQ:
+			x->op_count = 1;
+			cs_x86_op *op = &x->operands[0];
+			op->type =  X86_OP_REG;
+			op->reg = X86_REG_EFLAGS;
+			/*
+			if (id == X86_INS_PUSHF || id == X86_INS_POPF) {
```
Comment (radare) [inline libr/anal/p/anal_x86_cs.c]: Remove commented code

File: libr/anal/p/anal_x86_cs.c [hunk 3]
```diff
@@ -63,11 +63,58 @@ struct Getarg {
...
+			break;
+	}
+}
+
+static void opex(RStrBuf *buf, csh handle, cs_insn *insn, int mode) {
 	int i;
 	r_strbuf_init (buf);
 	r_strbuf_append (buf, "{");
 	cs_x86 *x = &insn->detail->x86;
+	if (x->op_count == 0){
```
Comment (radare) [inline libr/anal/p/anal_x86_cs.c]: Missing space before {

File: libr/anal/p/anal_x86_cs.c [hunk 4]
```diff
@@ -63,11 +63,58 @@ struct Getarg {
...
+		hidden_op(handle,insn,x,mode);
```
Comment (radare) [inline libr/anal/p/anal_x86_cs.c]: Space after every comma

Comment (radare) [reply libr/anal/p/anal_x86_cs.c]: Missing space before (

**Comment by x0urc3** [comment]
Reread code style and restyle

**Comment by radare** [comment]
Please fix the lgtm issue

**Comment by radare** [comment]
fix the lgtm alert please

**Comment by radare** [comment]
it was already using a global before but this is a bad idea, so we should avoid using globals at all

**Pull request merged**